Document how to use symfony/mailer for email SMTP relay

mstenta opened the pull request:
This changes our https://farmos.org/hosting/email/ documentation to recommend using `symfony/mailer` for SMTP email relays, instead of the [contrib SMTP module](drupal.org/project/smtp).

Kudos to @pcambra for turning me on to this new option! https://github.com/farmOS/farmOS/pull/837#issuecomment-2093359751

mstenta commented:
I came across this core issue, which is harmless but causes errors to appear in Apache logs: https://www.drupal.org/project/drupal/issues/3420372

There's a patch available, but I don't think we need to include it in farmOS. Hopefully it will be fixed in core soon.

mstenta commented:
On second thought, it's easy enough for us to include a patch, and risk is very low because it only touches the `SymfonyMailer.php` file. This will prevent the errors for anyone using `symfony/mailer`, and its easier to do this here in farmOS as part of our new recommendation than for each downstream project to patch Drupal core themselves.

I pushed a second commit to add this patch.

mstenta commented:
@paul121 and I isolated another Drupal core issue with the `SymfonyMailer` class: https://www.drupal.org/project/drupal/issues/3446368

I'm going to look into providing a patch for it...

paul121 commented:
Seeing these various bugs come up and considering that Drupal core still doesn't consider `symfony/mailer` integration to be complete I find it hard to make it the official suggestion here... there definitely will be significant benefits once the Email API is somewhat re-designed but until then, is there anything particularly bad about the `drupal/smtp` module?

mstenta commented:
I tend to agree we should wait on this PR. In the meantime I will test it out in Farmier and help to debug issues. Hopefully these two obvious bugs will be fixed in core quick so no new Drupal core patches will ever be needed in farmOS.

> Drupal core still doesn't consider [symfony/mailer](url) integration to be complete

That said, I think there's a difference between the "complete" integration as outlined in https://www.drupal.org/project/drupal/issues/1803948 and the simpler "SMTP relay" functionality that we currently recommend using the SMTP module for.

The change record (https://www.drupal.org/node/3369935) refers to the new `SymfonyMailer` plugin included with core as "experimental", so I'm not surprised there are still some bugs to work out. I'm optimistic that there won't be too many (the plugin itself is fairly simple), and we'll be able to recommend the SMTP relay capabilities long before the larger ambitions of refactoring Drupal's mail system happen.

mstenta commented:
> In the meantime I will test it out in Farmier and help to debug issues.

FWIW I have been running these two patches in Farmier-hosted instances for the past 2.5 months without any issue.

mstenta commented:
FYI: Drupal 11.3 adds a new experimental `mailer` module that leverages `symfony/mailer`. See change record for details: https://www.drupal.org/node/3519253

mstenta commented:
> I came across this core issue, which is harmless but causes errors to appear in Apache logs: https://www.drupal.org/project/drupal/issues/3420372

Update: this has been fixed in Drupal core 11.2.x!

The only other patch I'm currently applying is https://www.drupal.org/project/drupal/issues/3446368 - although that's only necessary if you send email to multiple addresses. The only farmOS code that might do that is data stream notifications, and only if you've configured a notification with multiple comma-separated addresses.

mstenta commented:
> The only other patch I'm currently applying is https://www.drupal.org/project/drupal/issues/3446368

This patch was just merged into 11.3.x, and 11.3.0 was released with it! :tada:

mstenta commented:
Rebased this onto `4.x`, removed the two commits for patching Drupal core, and force-pushed.

This is ready for final approval! I have been using this for almost 2 years without issue on many farmOS instances in production.

paul121 reviewed (approved):
Could we include a link to Drupal core docs on this? Not sure there is anything yet, the change record is OK but is already somewhat outdated https://www.drupal.org/node/3369935

mstenta commented:
> Could we include a link to Drupal core docs on this? Not sure there is anything yet...

The only thing I can find is this for Drupal CMS:

https://new.drupal.org/docs/drupal-cms/configuration/set-up-email-sending#step-2-configure-your-smtp-credentials-in-drupal

Drupal CMS ships with the [Symfony Mailer Lite](https://www.drupal.org/project/symfony_mailer_lite), so those docs assume you have that, and don't describe how to configure a simple SMTP relay like this PR is doing. The only thing I could find is the change record.

I'll add a link to the CR in our docs. Better than nothing, I suppose.
